src/libs/DateUtils.ts

```diff
@@ -728,6 +730,73 @@ function getLastBusinessDayOfMonth(inputDate: Date): number {
     return getDate(currentDate);
 }
 
+/**
+ * Returns a formatted date range from date 1 to date 2.
+ * Dates are formatted as follows:
+ * 1. When both dates refer to the same day: Mar 17
+ * 2. When both dates refer to the same month: Mar 17-20
+ * 3. When both dates refer to the same year: Feb 28 to Mar 1
+ * 4. When the dates are from different years: Dec 28, 2023 to Jan 5, 2024
+ */
+function getFormattedDateRange(date1: Date, date2: Date): string {
+    const {translateLocal} = Localize;
+
+    if (isSameDay(date1, date2)) {
+        // Dates are from the same day
+        return format(date1, 'MMM d');
+    }
+    if (isSameMonth(date1, date2)) {
+        // Dates in the same month and year, differ by days
+        return `${format(date1, 'MMM d')}-${format(date2, 'd')}`;
+    }
+    if (isSameYear(date1, date2)) {
+        // Dates are in the same year, differ by months
+        return `${format(date1, 'MMM d')} ${translateLocal('common.to').toLowerCase()} ${format(date2, 'MMM d')}`;
+    }
+    // Dates differ by years, months, days
+    return `${format(date1, 'MMM d, yyyy')} ${translateLocal('common.to').toLowerCase()} ${format(date2, 'MMM d, yyyy')}`;
+}
+
+/**
+ * Returns a formatted date range from date 1 to date 2 of a reservation.
+ * Dates are formatted as follows:
+ * 1. When both dates refer to the same day and the current year: Sunday, Mar 17
+ * 2. When both dates refer to the same day but not the current year: Wednesday, Mar 17, 2023
+ * 3. When both dates refer to the current year: Sunday, Mar 17 to Wednesday, Mar 20
+ * 4. When the dates are from different years or from a year which is not current: Wednesday, Mar 17, 2023 to Saturday, Jan 20, 2024
+ */
+function getFormattedReservationRangeDate(date1: Date, date2: Date): string {
+    const {translateLocal} = Localize;
+    if (isSameDay(date1, date2) && isThisYear(date1)) {
+        // Dates are from the same day
+        return format(date1, 'EEEE, MMM d, yyyy');
+    }
+    if (isSameDay(date1, date2)) {
+        // Dates are from the same day
+        return format(date1, 'EEEE, MMM d');
+    }
+    if (isSameYear(date1, date2) && isThisYear(date1)) {
+        // Dates are in the current year, differ by months
+        return `${format(date1, 'EEEE, MMM d')} ${translateLocal('common.conjunctionTo')} ${format(date2, 'EEEE, MMM d')}`;
+    }
+    // Dates differ by years, months, days or only by months but the year is not current
+    return `${format(date1, 'EEEE, MMM d, yyyy')} ${translateLocal('common.conjunctionTo')} ${format(date2, 'EEEE, MMM d, yyyy')}`;
+}
+
+/**
+ * Returns a formatted date of a transport mean departure.
```

Comment:
What does `mean` mean here? Maybe I'm sure reading it weirdly haha

Reply:
haha i don't even remb. simplified the comment